File: backend/app/api/workspaces.py
from fastapi import APIRouter, HTTPException, status

from app.models.workspace import Workspace, WorkspaceCreate
from app.services.workspace_service import workspace_service


router = APIRouter(
    prefix="/api/v1/workspaces",
    tags=["Workspaces"],
)


@router.post(
    "",
    response_model=Workspace,
    status_code=status.HTTP_201_CREATED,
)
def create_workspace(request: WorkspaceCreate) -> Workspace:
    """Create a workspace for a team or business function."""
    return workspace_service.create_workspace(request)


@router.get("", response_model=list[Workspace])
def list_workspaces() -> list[Workspace]:
    """Return all currently configured workspaces."""
    return workspace_service.list_workspaces()


@router.get("/{workspace_id}", response_model=Workspace)
def get_workspace(workspace_id: str) -> Workspace:
    """Return one workspace by its identifier."""
    workspace = workspace_service.get_workspace(workspace_id)

    if workspace is None:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="Workspace not found",
        )

    return workspace

File: frontend/src/api/workspaces.ts
import { apiClient } from "./client";
import { workspaceSchema } from "./schemas";
import type { Workspace, WorkspaceCreate } from "../types/investigation";

const BASE = "/api/v1/workspaces";

function parse(data: unknown): Workspace {
  return workspaceSchema.parse(data) as unknown as Workspace;
}

export async function listWorkspaces(): Promise<Workspace[]> {
  const response = await apiClient.get(BASE);
  return (response.data as unknown[]).map(parse);
}

export async function getWorkspace(workspaceId: string): Promise<Workspace> {
  const response = await apiClient.get(`${BASE}/${workspaceId}`);
  return parse(response.data);
}

export async function createWorkspace(request: WorkspaceCreate): Promise<Workspace> {
  const response = await apiClient.post(BASE, request);
  return parse(response.data);
}
